docs/modules/ROOT/pages/advanced-scripting.adoc

```diff
@@ -0,0 +1,749 @@
+----
+clc script output1.star -q -f json | jq '.Fruit'
+----
+Outputs:
+----
+"apple"
+"grapes"
+"pear"
+----
+
+
+Passing keyword arguments to `output` for column label and value may sometimes be limiting.
+Since only valid Starlark identifiers can be used as the column label.
+For example, running the following script will return an error:
+[source,python]
+----
+def main():
+    output(1=2)
+----
+Because of that, CLC also supports passing a dictionary to the `output` function.
+In that case, the dictionary must be passed as a positional argument and it must be the only argument to `output`.
+You can rewrite the following as follows and it will run just fine:
+[source,python]
+----
+def main():
+    output({1: 2})
+----
+
+The output function handles rows with missing values just fine.
+Here's a script that demonstrate that.
+Save it as `output2.star`:
+[source,python]
+----
+def main():
+    for i in range(3):
+        output({i: "OK"})
+----
+Running it outputs the following:
+----
+--------------
+ 0  | 1  | 2
+--------------
+ OK | -  | -
+ -  | OK | -
+ -  | -  | OK
+--------------
+----
+Different output formats handle missing values differently.
+JSON printer just skips the missing values:
+[source,bash]
+----
+clc script output2.star -f json
+----
+Outputs:
+----
+{"0":"OK"}
+{"1":"OK"}
+{"2":"OK"}
+----
+
+== Error Handling
+
+Whenever an error occurs, CLC stops the script and prints an error message.
+This is usually the safest way to handle the error.
+But there are times when you would like to try something which can result in an error and handle it without giving back the control.
+
+The `script` command supports `the --ignore-errors` flag that suppresses errors while running the script.
+You can use the `last_error()` function to get the raised error and act on it.
+
+Let's demonstrate that with an example using the `data_value` function.
+Passing a string to `data_value` ends with an error, but we are going to cover "why?" in another section.
+Save the following sctipt as `error.star`:
+
+[source,python]
+----
+def main():
+    v = data_value("foo")
+    print(v)
+----
+
+And run it as:
+
+[source,bash]
+----
+clc script error.star
+----
+
+You should get an error:
+----
+ERROR Error.star:2:19: argument 1 is expected to be 'serialization.Data', got 'string'
+----
+
+This time, run the script with the `--ignore-errors` flag:
+
+[source,bash]
+----
+clc script --ignore-errors error.star
+----
+
+You should get the following output:
+----
+None
+----
+
+The error in the `data_value` line is ignored and `v` is set to `None`.
+
+Let's now try to use the `last_error` function and act on errors.
+Save the following script as `error2.star`:
+
+[source,python]
+----
+def main():
+    v = data_value("foo")
+    if last_error():
+        v = "OK Caught the error."
+    print(v)
+----
+
+Run the script with the `--ignore-errors` flag:
+[source,bash]
+----
+clc script --ignore-errors error2.star
+----
+
+This time the output is:
+
```

Comment:
add the information on where it is caught and what to do - or remove the statement at the start about acting on errors

Reply:
I've added this:
> Since you have used the `--ignore-errors` flag, the program didn't end with an error in the `data_value` line but the error was recorded.
> Using the `last_error` function, the existence of an error was checked and `v` was set to `OK Caught the error.`.